### PR TITLE
Add optional parameter timeout to transcieve

### PR DESCRIPTION
**Pull request by smlu**
As discussed in PR #6, this PR add optional `timeout` parameter to `transcieve` method on Android.

### CODE REVIEW

**Comment by smlu** [comment]
Side Note: As already mentioned in PR [#6](https://github.com/nfcim/flutter_nfc_kit/pull/6#issuecomment-621768180) the `transcieve ` method should maybe be refactored to accept and return `UInt8List` raw APDU bytes besides hex string. This would make communication with smart card more efficient.

File: android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt [hunk 1]
```diff
@@ -33,6 +33,16 @@ class FlutterNfcKitPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
         private var pollingTimeoutTask: TimerTask? = null
         private var tagTechnology: TagTechnology? = null
+        private fun TagTechnology.transcieve(data: ByteArray, timeout: Int?) : ByteArray {
+            if(timeout != null) {
+                try {
+                    val timeoutMethod = this.javaClass.getMethod("setTimeout", Int::class.java)
+                    timeoutMethod.invoke(this, timeout)
+                } catch (ex: Throwable){}
+            }
+            val timeoutMethod = this.javaClass.getMethod("transceive", ByteArray::class.java)
```
Comment (Harry-Chen) [inline android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt:43]: maybe rename as `transceiveMethod`

**Comment by Harry-Chen** [comment]
> Side Note: As already mentioned in PR [#6](https://github.com/nfcim/flutter_nfc_kit/pull/6#issuecomment-621768180) the `transcieve ` method should maybe be refactored to accept and return `UInt8List` raw APDU bytes besides hex string. This would make communication with smart card more efficient.

LGTM, you could also do it in this PR

**Comment by Harry-Chen** [comment]
And I am requesting a review from @jiegec for the Swift part.

File: android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt (hunk 1 above)
Comment (smlu) [reply android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt:43]: Good catch, I'll fix it.

**Comment by smlu** [comment]
> 
> 
> > Side Note: As already mentioned in PR [#6](https://github.com/nfcim/flutter_nfc_kit/pull/6#issuecomment-621768180) the `transcieve ` method should maybe be refactored to accept and return `UInt8List` raw APDU bytes besides hex string. This would make communication with smart card more efficient.
> 
> LGTM, you could also do it in this PR


Ok, I'll make a new commit to this PR and refactor `transcieve` method to accept/return `UInt8List` object type besides hex string. 
I'm thinking to use generic type instead of `dynamic` type to do this. I believe it'll make the interface more clear and easier to use because the return type will be the same as input type.
Is it ok to go this way or do you prefer the `dynamic` type instead?

**Comment by Harry-Chen** [comment]
> > > Side Note: As already mentioned in PR [#6](https://github.com/nfcim/flutter_nfc_kit/pull/6#issuecomment-621768180) the `transcieve ` method should maybe be refactored to accept and return `UInt8List` raw APDU bytes besides hex string. This would make communication with smart card more efficient.
> > 
> > 
> > LGTM, you could also do it in this PR
> 
> Ok, I'll make a new commit to this PR and refactor `transcieve` method to accept/return `UInt8List` object type besides hex string.
> I'm thinking to use generic type instead of `dynamic` type to do this. I believe it'll make the interface more clear and easier to use because the return type will be the same as input type.
> Is it ok to go this way or do you prefer the `dynamic` type instead?

Sure, go ahead!

**Comment by smlu** [comment]
PR has been updated with refactored `transcieve` function to use generics for return type and for the type of parameter `capdu`. The generic type can be either of type `Uint8List` (bytes) or hex string. 
The change was tested of Android and iOS using biometric passport for test.

**Comment by Harry-Chen** [comment]
It's really considerate to match return data type with request type.

**Comment by Harry-Chen** [comment]
Someone told me that there is a typo in your PR 😆 

https://github.com/nfcim/flutter_nfc_kit/blob/46b688aa15747ab4149d6a1e955624ac2e18395c/android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt#L36

https://github.com/nfcim/flutter_nfc_kit/blob/46b688aa15747ab4149d6a1e955624ac2e18395c/android/src/main/kotlin/im/nfc/flutter_nfc_kit/FlutterNfcKitPlugin.kt#L112

Both `transceive` are misspelt.

I mean no blame of you, just feeling interesting that nobody (including me) had noticed that so far. Will fix it soon. 🔨

**Comment by smlu** [comment]
@Harry-Chen, my bad 😬.  It must have slipped through spellchecker. Sorry for this typo!

**Comment by Harry-Chen** [comment]
The typo is fixed in ff10d09. BTW I also noticed the title of this PR had the same typo. I have also corrected that. 🤣

**Pull request merged**